src/replication.c

```diff
@@ -1324,20 +1382,43 @@ check_for_command_completion(spec_t *spec, rcommon_cmd_t *rcomm_cmd, ISTGT_LU_CM
 		}
 	}
 
+	response_received = success + failure;
+	min_response = MAX(rcomm_cmd->replication_factor -
```

Comment:
min_response also depends on number of healthy and degraded replicas in the system when the responses are received.
For read IO, success response from one healthy replica is fine.
For write IO, success response from (consistency_factor) number of healthy replicas is fine.

For read IOs, below code (with comments incorporated) should be fine (we can re-arrange code to make this simple), even though, there is check for min_response which happens only for few IOs.
For write IOs, with this check for min_response, we are making the consistency factor as MAX(RF-CF+1, CF). We need to count success responses from healthy replicas, and success responses from degraded replicas to fix it. Think of case of 7 as RF, and 3 as CF.

Reply:
ok. I will modify write command handling.